src/app/clusters/ota-requestor/DefaultOTARequestorDriver.h

```diff
@@ -109,7 +109,10 @@ class DefaultOTARequestorDriver : public OTARequestorDriver
     uint16_t maxDownloadBlockSize  = 1024;
     // Maximum number of times to retry a BUSY OTA provider before moving to the next available one
     static constexpr uint8_t kMaxBusyProviderRetryCount = 3;
-    uint8_t mProviderRetryCount; // Track retry count for the current provider
+    // Track retry count for the current provider
+    uint8_t mProviderRetryCount = 0;
+    // Track query image retry count on invalid session error
+    uint8_t mInvalidSessionRetryCount = 0;
```

Comment:
Why do we need a second retry count? Since `mProviderRetryCount` is incremented in `SendQueryImage()` anyway, we'd always know how many times we've tried? We can just check `mProviderRetryCount < kMaxInvalidSessionRetries` in `HandleIdleStateEnter` directly?

Reply:
Done.

Reply:
Looks like `mProviderRetryCount` is not currently implemented correctly as a retry counter and needs to be fixed (in a separate PR).  Since provider retry could occur due to reasons other than invalid session, I would like to keep it as a separate variable for clarity.  If/when CASE invalidation/retry happens under the hood in the future such that we don't need to handle invalid session in the application layer, we can remove the session retry logic.